gammapy/background/ring.py

```diff
@@ -2,22 +2,236 @@
 """Ring background estimation.
 """
 from __future__ import absolute_import, division, print_function, unicode_literals
+from itertools import product
 import numpy as np
-from astropy.convolution import Ring2DKernel
+from astropy.convolution import Ring2DKernel, Tophat2DKernel
+import astropy.units as u
 from ..image import SkyImageList, SkyImage
+from ..image.utils import scale_cube_fft
 
 
 __all__ = [
+    'AdaptiveRingBackgroundEstimator',
     'RingBackgroundEstimator',
     'ring_r_out',
     'ring_area_factor',
     'ring_alpha',
 ]
 
 
-class RingBackgroundEstimator(object):
+class AdaptiveRingBackgroundEstimator(object):
+    """
+    Adaptive ring background algorithm.
+
+    This alogrithm extends the standard `RingBackground` method by adpating the
+    size of the ring to achieve a minimum on / off exposure ratio (alpha) in regions
+    where the area to estimate the background from is limited.
+
+    Here is an illustration of the method:
```

Comment:
Maybe put high-level docs like method illustrations in `docs` instead of having it in the API docs section?
(as you like, I'm not sure it's better ...)